patches/tModLoader/Terraria/Main.TML.cs

```diff
@@ -190,6 +190,11 @@ public partial class Main
 				ErrorReporting.FatalExit(Language.GetTextValue("tModLoader.ContentFolderNotFound"));
 			}
 
+			// Canary file, ensures that Terraria has updated to at least the version this tModLoader was built for
+			if (!File.Exists(Path.Combine(vanillaContentFolder, "Images", "Projectile_981.xnb"))) {
+				ErrorReporting.FatalExit(Language.GetTextValue("tModLoader.TerrariaOutOfDateMessage"));
```

Comment:
Is this needed given the build check ID below? Is there a possibility of someone having an incomplete Terraria install? If so, maybe this should just tell people to verify their terraria local files instead.